### PR TITLE
implement runtime api for disputes

### LINKED ISSUE #3159: Implement and integrate Dispute runtime APIs
blocked on #2947

1. Disputes Info: fetch on-chain data about active disputes
2. Rollback status: query the chain for a target to roll back the chain to, if any. This needs to be specified in the guide as well. 

Both these runtime APIs should be added to `v1::ParachainHost`, implemented in `runtime_api_impl::v1` using the disputes module, added to all runtimes, and integrated into the `RuntimeApiSubsystem`.

### PR DESCRIPTION
**Pull request by andresilva**
The actual implementation is just a placeholder as it's dependent on #2947 but all of the infrastructure is in place now.

Closes #3159.

TODO: add tests and update guide.

### CODE REVIEW

**Comment by rphmeier** [comment]
Reflecting bit further on the [`provisioner`](https://w3f.github.io/parachain-implementers-guide/node/utility/provisioner.html) algorithm we need to include disputes - we need information about the spam slots to know what the runtime will expect. The provisioner guide seems to think there's a `DisputeInfo` struct which probably looks like this:

```rust
struct DisputeInfo {
    spam_slots: Vec<SessionIndex, Vec<u32>>,
    max_spam_slots: u32,
    disputes: Vec<(SessionIndex, CandidateHash, DisputeState)>,
}
```

File: primitives/src/v1/mod.rs [hunk 1]
```diff
@@ -1216,6 +1220,17 @@ pub struct DisputeState<N = BlockNumber> {
...
 
+// FIXME
+#[cfg(feature = "std")]
+impl<N> MallocSizeOf for DisputeState<N> {
+	fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+		0
+	}
+	fn constant_size() -> Option<usize> {
+		Some(0)
+	}
```
Comment (ordian) [inline primitives/src/v1/mod.rs:1231]: this probably should probably not implemented, because it's not constant

**Comment by rphmeier** [comment]
superseded by new approach

File: primitives/src/v1/mod.rs (hunk 1 above)
Comment (serroopoint) [reply primitives/src/v1/mod.rs:1231]: Git

**Pull request closed without merging**